qiskit/dagcircuit/__init__.py

```diff
@@ -36,7 +36,18 @@
 
 .. autoexception:: DAGCircuitError
 .. autoexception:: DAGDependencyError
+
+Utilities
+=========
+
+.. autosummary::
+   :toctree: ../stubs/
+
+   BlockCollapser
+   BlockCollector
+   BlockSplitter
 """
+from .collect_blocks import BlockCollapser, BlockCollector, BlockSplitter
```

Comment:
Fwiw, this is a "high-level" part of `qiskit.dagcircuit` and depends on all the "low-level" bits (`DAGOpNode`, `DAGCircuit`, etc), so the cyclic import problems were mostly because Python needs to import things topologically from low-level to high-level.  Using specific imports (`from .dagnode import DAGOpNode`, e.g.) lets Python do things out of order, but the alternative would have been to leave `collect_blocks.py` exactly as is, but put this import line last.

No need to change anything here - we have this sort of thing all over the place - just commenting in case it helps in the future.

(For similar reasons, in this file `.exceptions` should be the first import, then the DAG nodes, then then DAGs.)